src/api/api_missions.nit

```diff
@@ -35,6 +36,40 @@ end
 
 class APIMission
 	super MissionHandler
+	super AuthHandler
+
+	redef fun post(req, res) do
+		var player = get_player(req, res)
+		if player == null then return
+		var mission = get_mission(req, res)
+		if mission == null then return
+		var post = req.post_args
+		if not post.has_key("source") then
+			res.error 400
+			print "No code provided for submission"
+			return
+		end
+		var code = post["source"]
+		if not post.has_key("lang") then
+			res.error 400
+			print "No lang provided for submission"
```

Comment:
That's a very good question, I still have no idea what's best yet.

One think I kept in mind was the way HackerRank keeps track of such infos: i.e. players choose their language/engine when submitting code.

One key difference here would be in the choice of engine and the attribution of stars.

Depending on the engine/language, there are differences in the size of the final binary and the time it takes to finish execution, therefore this must be taken into account when attributing bonuses.

The configuration file could have another attribute stating which languages are accepted for solving the problem, but there is still the star problem for which I have no real solution for now :(